go/plugins/postgresql/docstore.go

```diff
@@ -0,0 +1,157 @@
+package postgresql
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/firebase/genkit/go/ai"
+)
+
+type docStore struct {
+	engine PostgresEngine
+
+	tableName             string
+	schemaName            string
+	idColumn              string
+	metadataJsonColumn    string
+	contentColumn         string
+	embeddingColumn       string
+	metadataColumns       []string
+	ignoreMetadataColumns []string
```

Comment:
We shouldn't need ignoreMetadataColumns here. When we create the docstore, we will just set the metadataColumns we need